lib/misc.ml

```diff
@@ -36,77 +36,43 @@ let dot_name name_xxx =
   let base = Filename.chop_extension name_xxx in
   base ^ ".dot"
 
-let filebase f =
-  let f = Filename.basename f in
-  let b =
-    try Filename.chop_extension f
-    with Invalid_argument _ -> f in
-  b
-
+let filebase f = Filename.basename f |> Filename.remove_extension
 
 (****************)
 (* basic utils  *)
 (****************)
 
 let polymorphic_compare = compare
 
-external int_compare : int -> int -> int = "caml_int_compare"
-let int_eq (x:int) (y:int) = x == y
+let int_compare = Int.compare
+let int_eq = Int.equal
 let max_int (x:int) (y:int) = if x >= y then x else y
-let string_eq (s1:string) (s2:string) = (=) s1 s2
-
-let bool_eq b1 b2 = match b1,b2 with
-| (true,true)|(false,false) -> true
-| (false,true)|(true,false) -> false
-
-external identity : 'a -> 'a = "%identity"
+let string_eq = String.equal
+let bool_eq = Bool.equal
+let identity = Fun.id
 
 let ing _ = ()
 let ing2 _ _ = ()
 
 let not_found () = raise Not_found
 
-let is_none = function
-  | None -> true
-  | Some _ -> false
-
-let is_some = function
-  | None -> false
-  | Some _ -> true
-
-let as_some = function
-  | Some x -> x
-  | None -> assert false
-
-let proj_opt default = function
-  | None -> default
-  | Some x -> x
-
-let seq_opt f = function
-  | Some x -> f x
-  | None -> None
-
-let app_opt f = function
-  | None -> None
-  | Some x -> Some (f x)
+let is_none = Option.is_none
+let is_some = Option.is_some
+let as_some = Option.get
```

Comment:
This changes the behavior when its argument is None from assert to raise Invalid_argument but I think this makes more sense.